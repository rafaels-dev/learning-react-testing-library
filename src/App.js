import React, { useState, useEffect } from 'react';
import logo from './logo.svg';
import useLoading from './hooks/useLoading';
import './App.css';

function App() {

  const { loading, setLoading } = useLoading();

  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 500);
  }, [])

  if (loading) {
    return (
      <div id="loading" data-testid="loader">Carregando...</div>
    )
  }

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
