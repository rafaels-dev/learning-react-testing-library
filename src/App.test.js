import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import App from './App';

test('renders loading', () => {
  const { getByTestId } = render(<App />);
  const loadingElement = getByTestId('loader');
  expect(loadingElement).toBeInTheDocument();
  expect(loadingElement).toMatchSnapshot();
})

test('renders learn react link', async () => {
  const { getByText } = render(<App />);
  const linkElement = await waitForElement(
    () => getByText(/learn react/i)
  );
  expect(linkElement).toBeInTheDocument();
  expect(linkElement).toMatchSnapshot();
});
