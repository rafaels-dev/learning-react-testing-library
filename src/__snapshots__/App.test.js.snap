// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<a
  class="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React
</a>
`;

exports[`renders loading 1`] = `
<div
  data-testid="loader"
  id="loading"
>
  Carregando...
</div>
`;
